Framework/Kernel/inc/MantidKernel/Timer.h

```diff
@@ -40,5 +40,47 @@ class MANTID_KERNEL_DLL Timer {
 
 MANTID_KERNEL_DLL std::ostream &operator<<(std::ostream &, const Timer &);
 
+class MANTID_KERNEL_DLL CodeBlockTimer {
+public:
+  CodeBlockTimer() = delete;
+  CodeBlockTimer(const std::string &name, std::ostream &output);
+  ~CodeBlockTimer();
+
+private:
+  std::string name;
+  std::ostream &out;
+  std::chrono::time_point<std::chrono::system_clock> start;
+};
+
+class MANTID_KERNEL_DLL CodeBlockMultipleTimer {
+public:
+  class MANTID_KERNEL_DLL TimeAccumulator {
+  public:
+    TimeAccumulator() = delete;
+    TimeAccumulator(const std::string &name);
+
+  public:
+    void reset();
+    void increment(const double time_s);
+    double getElapsed() const;
+    size_t getNumberOfEntrances() const;
+    void report(std::ostream &out) const;
+
+  private:
+    std::string name;
```

Comment:
According to the [mantid c++ coding standards](https://developer.mantidproject.org/Standards/CPPStandards.html#naming-conventions), these should be `m_name`, `m_elapsed_sec`, and `m_number_of_entrances`. I added the extra characters to the elapsed time to make it slightly more explicit.

Reply:
I have updated all class member names and changed `s` to `sec`.